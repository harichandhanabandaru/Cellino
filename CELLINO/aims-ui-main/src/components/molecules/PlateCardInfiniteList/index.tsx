import AutoSizer from "react-virtualized-auto-sizer";
import InfiniteLoader from "react-window-infinite-loader";
import { List } from "@mui/material";
import { FixedSizeList } from "react-window";
import memoize from "memoize-one";
import CardBox from "../../organisms/plateCard";
import React from "react";
import { Plate } from "../../../constants/types";

const THRESHOLD = 15;

const Item = ({
  data,
  index,
  style,
}: {
  index: number;
  style: React.CSSProperties;
  data: { phaseData: Plate[]; phase: string | undefined };
}) => {
  const requiredData = data.phaseData;
  const currentItem = requiredData[index];
  const phase = data.phase;
  return (
    currentItem && (
      <div style={style}>
        <CardBox cardDetails={currentItem} key={index} phaseName={phase} />
      </div>
    )
  );
};

const createRequiredItemData = memoize((items) => items);

interface PlateCardInfiniteListProps {
  data: Array<any>;
  loadNextPage: (startIndex: number, stopIndex: number) => void;
  totalElements: number;
  last: boolean;
  itemSize: number;
  phaseName?: string;
}

function PlateCardInfiniteList({
  data,
  loadNextPage,
  totalElements,
  last,
  itemSize,
  phaseName,
}: PlateCardInfiniteListProps) {
  const isItemLoaded = (index: number) => last || index < data.length;

  const requiredItemData = createRequiredItemData(data);

  return (
    <AutoSizer>
      {({ height, width }) => (
        <InfiniteLoader
          isItemLoaded={isItemLoaded}
          itemCount={totalElements}
          loadMoreItems={loadNextPage}
          threshold={THRESHOLD}
        >
          {({ onItemsRendered, ref }) => (
            <List disablePadding>
              <FixedSizeList
                height={height}
                width={width}
                itemCount={requiredItemData.length}
                itemSize={itemSize}
                itemData={{
                  phaseData: requiredItemData,
                  phase: phaseName,
                }}
                onItemsRendered={onItemsRendered}
                ref={ref}
              >
                {Item}
              </FixedSizeList>
            </List>
          )}
        </InfiniteLoader>
      )}
    </AutoSizer>
  );
}

export default PlateCardInfiniteList;
