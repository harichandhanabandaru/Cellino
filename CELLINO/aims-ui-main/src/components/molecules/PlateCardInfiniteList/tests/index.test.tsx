import { render } from "@testing-library/react";
import PlateCardInfiniteList from "../index";

const ITEM_SIZE = 125;

describe("plate card infinte lsi tests", () => {
  it("should match the palte card infinite list render", () => {
    const loadMoreItems = jest.fn();
    const { container } = render(
      <PlateCardInfiniteList
        data={[
          {
            currentPhaseId: "038afa29-6c07-4476-aacf-d1291833c66a",
            id: "8",
            plateName: "x1412",
            processStatus: "",
            runId: "8beb169a-f116-11ec-8ea0-0242ac120002",
          },
        ]}
        loadNextPage={loadMoreItems}
        totalElements={1}
        last={true}
        itemSize={ITEM_SIZE}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
