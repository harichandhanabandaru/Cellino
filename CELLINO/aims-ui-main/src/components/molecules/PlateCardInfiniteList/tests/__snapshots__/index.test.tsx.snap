// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plate card infinte lsi tests should match the palte card infinite list render 1`] = `
<div>
  <div
    style="overflow: visible; height: 0px; width: 0px;"
  />
  <div
    class="resize-triggers"
  >
    <div
      class="expand-trigger"
    >
      <div
        style="width: 1px; height: 1px;"
      />
    </div>
    <div
      class="contract-trigger"
    />
  </div>
</div>
`;
